src/ethereum-canister/src/lib.rs

```diff
@@ -69,8 +73,29 @@ async fn get_block_number() -> Nat {
     head_block_num.into()
 }
 
+#[query]
+async fn get_gas_price() -> U256 {
+    let helios = helios::client();
+
+    let gas_price = helios.get_gas_price().await.expect("get_gas_price failed");
+
+    gas_price.into()
+}
+
+#[update]
```

Comment:
> and both of them send an RPC call

`get_gas_price` doesn't do an RPC call so it can stay as a query. Unfortunately RPC calls leads to `call_with_payment` that changes the canister's balance and forbid using `query` there.